src/webgpu/util/texture/texelData.ts

```diff
@@ -0,0 +1,359 @@
+  componentOrder: TexelComponent[];
+  componentInfo: TexelComponentInfo;
+  getBytes(components: { [c in TexelComponent]?: number }): ArrayBuffer;
+}
+
+class TexelDataRepresentationImpl implements TexelDataRepresentation {
+  private constructor(
+    private readonly format: GPUTextureFormat,
+    readonly componentOrder: TexelComponent[],
+    readonly componentInfo: TexelComponentInfo,
+    private readonly sRGB: boolean = false
+  ) {}
+
+  static Create(format: GPUTextureFormat): TexelDataRepresentationImpl {
+    switch (format) {
+      case C.TextureFormat.RGB10A2Unorm: {
+        const write10 = (n: number) => ({
+          value: floatAsNormalizedInteger(n, 10, false),
+          type: TexelWriteType.Uint,
+        });
+        const write2 = (n: number) => ({
+          value: floatAsNormalizedInteger(n, 2, false),
+          type: TexelWriteType.Uint,
+        });
+
+        return new TexelDataRepresentationImpl(format, kComponentOrderRGBA, {
+          R: { bitLength: 10, write: write10 },
+          G: { bitLength: 10, write: write10 },
+          B: { bitLength: 10, write: write10 },
+          A: { bitLength: 2, write: write2 },
+        });
+      }
+
+      case C.TextureFormat.RGB11B10Float: {
+        const write11 = (n: number) => ({
+          value: float32ToFloatBits(n, 0, 5, 6, 15),
+          type: TexelWriteType.Uint,
+        });
+        const write10 = (n: number) => ({
+          value: float32ToFloatBits(n, 0, 5, 5, 15),
+          type: TexelWriteType.Uint,
+        });
+
+        return new TexelDataRepresentationImpl(format, kComponentOrderRGB, {
+          R: { bitLength: 11, write: write11 },
+          G: { bitLength: 11, write: write11 },
+          B: { bitLength: 10, write: write10 },
+        });
+      }
+
+      case C.TextureFormat.Depth32Float: {
+        return new TexelDataRepresentationImpl(format, [TexelComponent.Depth], {
+          Depth: {
+            bitLength: 32,
+            write: (n: number) => ({
+              value: n,
+              type: TexelWriteType.Float,
+            }),
+          },
+        });
+      }
+
+      case C.TextureFormat.Depth24Plus: {
+        return new TexelDataRepresentationImpl(format, [TexelComponent.Depth], {
+          Depth: {
+            bitLength: 32,
+          },
+        });
+      }
+
+      case C.TextureFormat.Depth24PlusStencil8: {
+        return new TexelDataRepresentationImpl(
+          format,
+          [TexelComponent.Depth, TexelComponent.Stencil],
+          {
+            Depth: {
+              bitLength: 24,
+            },
+            Stencil: {
+              bitLength: 8,
+              write: (n: number) => ({
+                value: (assertInIntegerRange(n, 8, false), n),
+                type: TexelWriteType.Uint,
+              }),
+            },
+          }
+        );
+      }
+
+      default:
+        break;
+    }
+
+    const m = format.match(/([rgba]+)(\d+)(snorm|unorm|uint|sint|float)(-srgb)?/);
```

Comment:
Perhaps we should put all this metadata in the capability_info instead of trying to compute it?

Reply:
Maybe? But it would all be duplicate information, and that table is getting huge. Seems odd to have:

`'r32uint': { components: ['r'], bits: 32, type: 'uint' }`
^ This representation also doesn't work for all the formats (like rg11b10float) so it would probably be more complex or have even more fields

Reply:
What about about a new table that's just a hardcoded version of kRepresentationCache? Seems less complex than this logic.

Reply:
Separate table is a good idea; did that in the latest update